### ISSUE TITLE
Input issues (joypad support, spacebar jump)

### ISSUE THREAD

**Issue by HerbFargus** [radgeRayden/future-mari0#1]
Just posting an issue for official reference for the space bar needing to be reconfigured in order to jump along with a request for joypad support.

**Comment by radgeRayden** [comment]
The jump problem is caused by the game expecting a ' ' constant for space bar by default, while it has been changed to 'space' as of 0.10.0. For now it's possible to make the game accept the right constant since the function the options menu uses to capture input has been rewired to accept it. I'm looking for the code responsible for game input, and then I'll be able to modify internal behaviour.

tl;dr the workaround for this is to edit input options on the menu for the right keys. This will work out of the box at some point.

**Comment by adid4s** [comment]
When i try to map the input,  The game recognizes my wireless 360 controller buttons which is a good sign,  but just crashes when I try to start the game afterwards,  joy pad compatibility would be amazing,  hope to see it happen one day

**Comment by adid4s** [comment]
When i try to map the input,  The game recognizes my wireless 360 controller buttons which is a good sign,  but just crashes when I try to start the game afterwards,  joy pad compatibility would be amazing,  hope to see it happen one day

**Comment by adid4s** [comment]
When i try to map the input,  The game recognizes my wireless 360 controller buttons which is a good sign,  but just crashes when I try to start the game afterwards,  joy pad compatibility would be amazing,  hope to see it happen one day

**Comment by adid4s** [comment]
When i try to map the input,  The game recognizes my wireless 360 controller buttons which is a good sign,  but just crashes when I try to start the game afterwards,  joy pad compatibility would be amazing,  hope to see it happen one day

**Comment by radgeRayden** [comment]
Since two separate problems are being reported, I'll close this issue because the spacebar problem has just been fixed and will open another one about the gamepad crash.
The fix only affects new users / people with no config files.

**Comment by radgeRayden** [comment]
@adid4s as of the last commit gamepads are supported by the game. Please test it out and report if it doesn't work at some point.

**Comment by HerbFargus** [comment]
Just as a reference for anyone else updating from the first version: 

You need to delete 

```
/home/pi/.local/share/love/mari0/options.txt
```

and reboot before configurations will work. 

Also gamepad navigation doesnt work in the menu but it does work in the game.

**Comment by radgeRayden** [comment]
correction: the reboot is only necessary if you're on a raspberry pi

**Comment by adid4s** [comment]
How exactly would I update the game,  simply delete and re-download? Or is there an easier way,  and thanks @radgeRayden for your work I will be trying it out as soon as I'm home from work

**Comment by HerbFargus** [comment]
Just delete the current game and replace it with the latest release: https://github.com/radgeRayden/future-mari0/releases ill look at updating the retropie script when I've got some time.

**Comment by adid4s** [comment]
Thanks herb
